### PR TITLE
optstatus valid for IRCs

### PR DESCRIPTION
**Pull request by LaTruelle**
optstatus array should be properly updated now, and more cleanly.

### CODE REVIEW

**Comment by LaTruelle** [comment]
The Travis build failed because of an ORCA test on CO2 freqs...

**Comment by langner** [comment]
FTR this addresses #211 (doesn't link in the title for some reason).

**Comment by langner** [comment]
> The Travis build failed because of an ORCA test on CO2 freqs...

I believe those are being addressed by Eric in #426.

File: src/cclib/parser/gaussianparser.py [hunk 1]
```diff
@@ -821,10 +821,11 @@ def extract(self, inputfile, line):
         # Matches "Step number  123", "Pt XX Step number 123" and "PtXXX Step number 123"
         if " Step number" in line:
             step = int(line.split()[line.split().index('Step') + 2])
+            if not hasattr(self, "optstatus"):
+                self.optstatus = []
+            self.optstatus.append(data.ccData.OPT_UNKNOWN)
```
Comment (langner) [inline src/cclib/parser/gaussianparser.py]: Could you combine this and the next two lines into an if/else block?

File: src/cclib/parser/gaussianparser.py [hunk 2]
```diff
@@ -1730,6 +1723,48 @@ def extract(self, inputfile, line):
...
+        # Pt 40 Step number   2 out of a maximum of  20
+        # Modified Bulirsch-Stoer Extrapolation Cycles:
+        #    EPS =    0.000010000000000
+        # Maximum DWI energy   std dev =  0.000007300 at pt    31
+        # Maximum DWI gradient std dev =  0.085197906 at pt    59
+        # CORRECTOR INTEGRATION CONVERGENCE:
+        #   Recorrection delta-x convergence threshold:    0.010000
+        #   Delta-x Convergence NOT Met
+        # IRC-IRC-IRC-IRC-IRC-IRC-IRC-IRC-IRC-IRC-IRC-IRC-IRC-IRC-IRC-IRC-IRC-IRC
+
```
Comment (langner) [inline src/cclib/parser/gaussianparser.py]: nit: please remove this blank line.

File: src/cclib/parser/gaussianparser.py [hunk 3]
```diff
@@ -1730,6 +1723,48 @@ def extract(self, inputfile, line):
...
+        if line[3:22] == "Delta-x Convergence":
+            if line[23:30] == "NOT Met":
+                self.optstatus[-1] = data.ccData.OPT_UNCONVERGED
+            elif line[23:26] == "Met":
+                self.optstatus[-1] = data.ccData.OPT_DONE
```
Comment (langner) [inline src/cclib/parser/gaussianparser.py]: Is there any else action?

File: src/cclib/parser/gaussianparser.py (hunk 1 above)
Comment (LaTruelle) [reply src/cclib/parser/gaussianparser.py]: Done in 81bc37874361714c77c80b69ac519114b4783fbb.

File: src/cclib/parser/gaussianparser.py (hunk 3 above)
Comment (LaTruelle) [reply src/cclib/parser/gaussianparser.py]: Not that I can think of... it is actually only changing the statuswhen necessary.

**Comment by LaTruelle** [comment]
tests need updating for the new `OPT_XXX` values. Do you have pointers for this?

I will do the commit cleaning ASAP.

**Comment by langner** [comment]
> tests need updating for the new OPT_XXX values. Do you have pointers for this?

There are tests for `optstatus` in the unit test for[ geometry optimizations](https://github.com/cclib/cclib/blob/master/test/data/testGeoOpt.py) and [scans](https://github.com/cclib/cclib/blob/master/test/data/testScan.py). But, I think you'll need to update them in a non-trivial way. Let us know if you have any issues.

**Comment by LaTruelle** [comment]
I did a rebase. Can you tell me if it worked? It displays strangely on github but should be fine anyway. I will have a go at the tests and let you know.

**Comment by ATenderholt** [comment]
It doesn't look like the rebase worked since it's still showing 21 commits. Did you do a ```git push --force```?

**Comment by LaTruelle** [comment]
OK now it is a huge mess... I redid the rebase and force pushed, but it included for some reason a bunch of previous commits in the process (I suspect the `git rebase -i HEAD~21` to have messed up because of the previous failed rebase). Do you have pointers for how to recover from this? Start again from scratch?

**Comment by langner** [comment]
Yes, that looks very messed up. You can undo it, but it'll be a chore. You're best bet is to start a new branch and cherry pick your way into a clean history on top of master.

If you have further issues I can do this for you. Let me know.

**Comment by LaTruelle** [comment]
@langner I tried a few tings but to no avail. You have probably more experience in this so please feel free to sort this mess out. Otherwise if that make things easier I can also remove my fork, close the PR and start from scratch, but it will also mess up all references to this PR in #211 and other issues.

**Comment by langner** [comment]
OK, I cherry-picked your commit and did some follow-up work in #432. Closing this PR meanwhile. Thanks for running so far with this!

**Pull request closed without merging**